Tests/WitsmlExplorer.Api.Tests/Workers/SpliceLogsWorkerTests.cs

```diff
@@ -0,0 +1,540 @@
+            var job = CreateSpliceLogsJob(logUids);
+
+            WitsmlLogs logHeaders = CreateSampleLogHeaders(logUids, indexType, new int[] { 0, 5 }, new int[] { 4, 9 });
+            WitsmlLogs logData = CreateSampleLogData(logUids, indexType, new int[] { 0, 5 }, new int[] { 4, 9 });
+
+            var expectedMnemonics = logHeaders.Logs.First().LogCurveInfo.Select(lci => lci.Mnemonic);
+            var expectedUnits = logHeaders.Logs.First().LogCurveInfo.Select(lci => lci.Unit);
+
+            WitsmlLogs capturedLogHeader = null;
+            WitsmlLogs capturedLogData = null;
+            SetupClient(_witsmlClient, logHeaders, logData, (log) => capturedLogHeader = log, (log) => capturedLogData = log);
+
+            var (workerResult, refreshAction) = await _worker.Execute(job);
+
+            Assert.True(workerResult.IsSuccess);
+            Assert.NotNull(refreshAction);
+
+            // Verify that the new log has the correct header
+            Assert.NotNull(capturedLogHeader);
+            Assert.Single(capturedLogHeader.Logs);
+            WitsmlLog newLogHeader = capturedLogHeader.Logs.First();
+            Assert.Equal(_newLogUid, newLogHeader.Uid);
+            Assert.Equal(_newLogName, newLogHeader.Name);
+            Assert.Equal(_wellUid, newLogHeader.UidWell);
+            Assert.Equal(_wellboreUid, newLogHeader.UidWellbore);
+
+            Assert.Equal(expectedMnemonics, newLogHeader.LogCurveInfo.Select(lci => lci.Mnemonic));
+            Assert.Equal(expectedUnits, newLogHeader.LogCurveInfo.Select(lci => lci.Unit));
+
+            // Verify that the new log has the correct data
+            Assert.NotNull(capturedLogData);
+            Assert.Single(capturedLogData.Logs);
+            WitsmlLog newLogDataHeader = capturedLogData.Logs.First();
+            Assert.Equal(_newLogUid, newLogDataHeader.Uid);
+            Assert.Equal(_wellUid, newLogDataHeader.UidWell);
+            Assert.Equal(_wellboreUid, newLogDataHeader.UidWellbore);
+            WitsmlLogData newLogData = newLogDataHeader.LogData;
+            Assert.Equal(expectedMnemonics, newLogData.MnemonicList.Split(','));
+            Assert.Equal(expectedUnits, newLogData.UnitList.Split(','));
+            Assert.Equal(logData.Logs.First().LogData.Data.Select(d => d.Data), newLogData.Data.Select(d => d.Data).Take(5));
+            Assert.Equal(logData.Logs.Last().LogData.Data.Select(d => d.Data), newLogData.Data.Select(d => d.Data).TakeLast(5));
+        }
+
+        [Theory]
+        [InlineData(WitsmlLog.WITSML_INDEX_TYPE_MD)]
+        [InlineData(WitsmlLog.WITSML_INDEX_TYPE_DATE_TIME)]
+        public async Task Execute_SomeOverlap_KeepsFirstOnOverlap(string indexType)
+        {
+            string[] logUids = { "log1Uid", "log2Uid" };
+            var job = CreateSpliceLogsJob(logUids);
+
+            WitsmlLogs logHeaders = CreateSampleLogHeaders(logUids, indexType, new int[] { 0, 5 }, new int[] { 9, 14 });
```

Comment:
It might be useful to move the startIndexNum and endIndexNum lists into variables, to [prevent the need for the reader of the test to inspect the production code in order to figure out what makes the value special](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#why-3).